x-pack/legacy/plugins/lens/public/id_generator/mock.ts

```diff
@@ -0,0 +1,10 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+export function mockGeneratedIds(generator: { generateId: () => string }, ...ids: string[]) {
+  const mock = jest.spyOn(generator, 'generateId');
+  ids.forEach(id => mock.mockReturnValueOnce(id));
+}
```

Comment:
Can you take advantage of jest's ability to replace modules with manual mocks? https://jestjs.io/docs/en/manual-mocks

Here's what I was thinking:

In `id_generator/__mocks__/index.ts`:

```typescript
export function generateId() {
  return jest.fn();
}
```

In your test code:

```typescript
import { generateId } from '../id_generator';

jest.mock('../id_generator');

  test('allows adding y dimensions', () => {
    (generateId as jest.Mock).mockReturnValueOnce('zed');
```

Reply:
That looks much nicer. I'll give it a shot.